runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/FarmingTracker.java

```diff
@@ -121,20 +128,52 @@ public boolean updateData(WorldPoint location)
 				if (storedValue != null)
 				{
 					String[] parts = storedValue.split(":");
-					if (parts.length == 2 && parts[0].equals(strVarbit))
+					if (parts.length == 2)
 					{
-						long unixTime = 0;
-						try
+						if (parts[0].equals(strVarbit))
 						{
-							unixTime = Long.parseLong(parts[1]);
+							long unixTime = 0;
+							try
+							{
+								unixTime = Long.parseLong(parts[1]);
+							}
+							catch (NumberFormatException e)
+							{
+								// ignored
+							}
+							if (unixTime + (5 * 60) > unixNow && unixNow + 30 > unixTime)
+							{
+								continue;
+							}
 						}
-						catch (NumberFormatException e)
+						else if (!newRegionLoaded)
 						{
-							// ignored
-						}
-						if (unixTime + (5 * 60) > unixNow && unixNow + 30 > unixTime)
-						{
-							continue;
+							PatchState previousPatchState = patch.getImplementation().forVarbitValue(Integer.parseInt(parts[0]));
+							PatchState currentPatchState = patch.getImplementation().forVarbitValue(client.getVar(varbit));
+							int patchTickRate = previousPatchState.getTickRate();
+
+							if (isObservedGrowthTick(previousPatchState, currentPatchState))
+							{
+								Integer storedOffsetPrecision = configManager.getRSProfileConfiguration(TimeTrackingConfig.CONFIG_GROUP, TimeTrackingConfig.FARM_TICK_OFFSET_PRECISION, int.class);
+								Integer storedOffsetMins = configManager.getRSProfileConfiguration(TimeTrackingConfig.CONFIG_GROUP, TimeTrackingConfig.FARM_TICK_OFFSET, int.class);
+
+								int offsetMins = (int) Math.abs(((Instant.now().getEpochSecond() / 60) % patchTickRate) - patchTickRate);
+								log.debug("Farming: Observed an exact growth tick. Offset is: {} from a {} minute tick", offsetMins, patchTickRate);
+
+								if (storedOffsetMins != null && storedOffsetMins != 0 && offsetMins != storedOffsetMins % patchTickRate)
+								{
+									WorldPoint playerLocation = client.getLocalPlayer().getWorldLocation();
+									log.error("Farming: Offset error! Observed new offset of {}, previous observed offset was {} ({}) Player Loc:{}", offsetMins, storedOffsetMins, storedOffsetMins % patchTickRate, ((playerLocation != null) ? playerLocation.toString() : ""));
```

Comment:
```suggestion
									log.error("Farming: Offset error! Observed new offset of {}, previous observed offset was {} ({}) Player Loc:{}",
										offsetMins, storedOffsetMins, storedOffsetMins % patchTickRate, playerLocation));
```

playerLocation will never be null, and even if it could be this would be better